keanu-python/keanu/algorithm/_proposal_distribution.py

```diff
@@ -0,0 +1,37 @@
+from typing import List, Any, Callable, Dict
+
+from py4j.java_gateway import java_import
+
+from keanu.base import JavaObjectWrapper
+from keanu.context import KeanuContext
+from keanu.tensor import Tensor
+from keanu.vartypes import numpy_types
+
+k = KeanuContext()
+
+java_import(k.jvm_view(), "io.improbable.keanu.algorithms.mcmc.proposal.GaussianProposalDistribution")
+java_import(k.jvm_view(), "io.improbable.keanu.algorithms.mcmc.proposal.PriorProposalDistribution")
+
+proposal_distribution_types: Dict[str, Callable] = {
+    "gaussian": k.jvm_view().GaussianProposalDistribution,
+    "prior": k.jvm_view().PriorProposalDistribution,
+}
+
+
+class ProposalDistribution(JavaObjectWrapper):
+
+    def __init__(self, type_: str, sigma: numpy_types = None, listeners: List[Any] = []) -> None:
```

Comment:
The default is set on the Java side: it's "prior", although that is likely to change since it's not a very good proposal. So I don't want to implement a default on the python side as that could lead to them getting out of sync.

My intention was to allow you to specify no type on the python side, and it just works - but I haven't implemented that yet, so I will do that still.